src/cargo/util/toml/mod.rs

```diff
@@ -334,7 +334,7 @@ pub fn prepare_for_publish(
         dev_dependencies2: None,
```

Comment:
W8 might move every `map_deps` up to have their own local variables, so that

* In `map_feature` we won't repeat ourselves filtering dev-deps which we've done to `dev_dependencies` here.
* We can create a map of all dependencies-to-publish so `map_feature` knows which is safe to strip.

Reply:
A `strip_features` was added and is performed on the `TomlManifest` after the mapping, having the same effect